src/Controller/Component/AjaxComponent.php

```diff
@@ -88,11 +88,14 @@ protected function _respondAsAjax() {
 			$this->Controller->set('_message', $message);
 		}
 
-		$serializeKeys = ['_message'];
-		if (!empty($this->Controller->viewVars['_serialize'])) {
-			$serializeKeys = array_merge($serializeKeys, $this->Controller->viewVars['_serialize']);
+		// If _serialize is true, *all* viewVars will be serialized; no need to add _message.
+		if ($this->Controller->viewVars['_serialize'] !== true) {
+			$serializeKeys = ['_message'];
+			if (!empty($this->Controller->viewVars['_serialize'])) {
+				$serializeKeys = array_merge($serializeKeys, $this->Controller->viewVars['_serialize']);
+			}
+			$this->Controller->set('_serialize', $serializeKeys);
 		}
-		$this->Controller->set('_serialize', $serializeKeys);
```

Comment:
Having return early is one of the most useful code practices. I dont know any sane person who would argue against that.